src/sentry/migrations/0374_backfill_monitor_environments.py

```diff
@@ -0,0 +1,95 @@
+# Generated by Django 2.2.28 on 2023-03-07 17:55
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+CREATE_MONITORENV_QUERY = """
+
+"""
+
+UPDATE_QUERY = """
+    UPDATE sentry_monitorcheckin
+    SET monitor_environment_id = data.monitor_environment_id
+    FROM (VALUES %s) AS data (id, monitor_environment_id)
+    WHERE sentry_monitorcheckin.id = data.id"""
+
+
+def backfill_monitor_checkins(apps, schema_editor):
+    MonitorCheckIn = apps.get_model("sentry", "MonitorCheckIn")
+    MonitorEnvironment = apps.get_model("sentry", "MonitorEnvironment")
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        MonitorCheckIn.objects.all().select_related("monitor")
+    )
+    cursor = connection.cursor()
+    batch = []
+    for monitor_checkin in queryset:
+        if monitor_checkin.monitor_environment:
+            pass
+
+        monitor = monitor_checkin.monitor
+
+        environment = Environment.objects.get_or_create(
+            name=DEFAULT_ENVIRONMENT_NAME, organization_id=monitor.organization_id
+        )[0]
+
+        EnvironmentProject.objects.get_or_create(
+            project_id=monitor_checkin.project_id, environment=environment, is_hidden=None
+        )
```

Comment:
Is it cool to call `get_or_create` in the loop like this? Or should it be batched and broken up somehow?

Reply:
This seems fine to me. Maybe @wedamija has more opinions here.

Reply:
We'll need to batch these calls, otherwise this migration will take a really long time. At the moment, `MonitorCheckin` has around 39mm rows, so that means for every checkin we're going to be making 4-5 io calls, which will slow this down a lot